lib/cpm/shell/log.sh

```
@@ -0,0 +1,39 @@
+#
+# Usage: source this, then call log_run at end of any gate.
+#   log_run "check-fast" 0    # gate name, exit code
+#   log_run "pre-commit" 1    # failed
+
+LOG_FILE=".tmp/checks.log"
+mkdir -p .tmp
+
+log_run() {
+  local gate="$1"
+  local exit_code="${2:-0}"
+  local ts=$(date +%Y-%m-%dT%H:%M:%S%z)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)